packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -100,10 +107,73 @@ function emitMixedTypeAnnotation(
   });
 }
 
+function getTypeScriptMemberType(maybeEnumDeclaration: $FlowFixMe) {
+  if (maybeEnumDeclaration.members[0].initializer) {
+    return maybeEnumDeclaration.members[0].initializer.type
+      .replace('NumericLiteral', 'NumberTypeAnnotation')
+      .replace('StringLiteral', 'StringTypeAnnotation');
+  }
+
+  return 'StringTypeAnnotation';
+}
+
+function getFlowMemberType(maybeEnumDeclaration: $FlowFixMe) {
+  return maybeEnumDeclaration.body.type
+    .replace('EnumNumberBody', 'NumberTypeAnnotation')
+    .replace('EnumStringBody', 'StringTypeAnnotation');
+}
+
+function translateDefault(
+  hasteModuleName: string,
+  typeAnnotation: $FlowFixMe,
+  types: TypeDeclarationMap,
+  nullable: boolean,
+  language: ParserType,
+  parser: Parser,
+): Nullable<NativeModuleEnumDeclaration> {
+  const maybeEnumDeclaration =
+    types[parser.nameForGenericTypeAnnotation(typeAnnotation)];
+
+  if (
+    maybeEnumDeclaration &&
+    (maybeEnumDeclaration.type === 'TSEnumDeclaration' ||
+      maybeEnumDeclaration.type === 'EnumDeclaration')
+  ) {
```

Comment:
can you create a function in the `parser.js` interface called:
```
isEnumDeclaration(maybeEnumDeclaration: $FlowFixMe):bollean
```

which checks, for Flow, `maybeEnumDeclaration.type === 'EnumDeclaration'` and for typescript `maybeEnumDeclaration.type === 'TSEnumDeclaration'`?

In this way, we can replace this if with:

```ts
if (maybeEnumDeclaration && parser.isEnumDeclaration(maybeEnumDeclaration))

Reply:
Yeah updated, it's much cleaner now